### PR TITLE
Rename configuration parameters, update README

### LINKED ISSUE #13: Rename configuration values
### Feature description

Our convention is to use camel case for naming configuration parameter names. For example, access-key-id should be accessKeyId etc.

### PR DESCRIPTION
**Pull request by hariso**
Fixes #13.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,12 +3,26 @@
...
+| `s3.region`            | S3 region                                           | true     | -             | "us-east-1"                                                                                                                                                                                                |
+| `catalog.propertyName` | Set a catalog property with the name `propertyName` | true     | -             | {"catalog.uri": "http://localhost:8181"}                                                                                                                                                                   |
+| `catalog.catalog-impl` | Catalog implementation to be used                   | true     | -             | "org.apache.iceberg.rest.RESTCatalog" <br/> Possible values: <br/> - "org.apache.iceberg.hadoop.HadoopCatalog" <br/> - "org.apache.iceberg.jdbc.JdbcCatalog" <br/> - "org.apache.iceberg.rest.RESTCatalog" |
```
Comment (maha-hajja) [inline README.md:28]:
```suggestion
| `catalog.catalog-impl` | Catalog implementation to be used                   | true     | -             | `"org.apache.iceberg.rest.RESTCatalog"` <br/> Possible values: <br/> - `"org.apache.iceberg.hadoop.HadoopCatalog"` <br/> - `"org.apache.iceberg.jdbc.JdbcCatalog"` <br/> - `"org.apache.iceberg.rest.RESTCatalog"` |
```

File: README.md [hunk 2]
```diff
@@ -3,12 +3,26 @@
...
+| `s3.secretAccessKey`   | S3 Secret Access Key                                | true     | -             | "exampleSecretKey"                                                                                                                                                                                         |
+| `s3.region`            | S3 region                                           | true     | -             | "us-east-1"                                                                                                                                                                                                |
+| `catalog.propertyName` | Set a catalog property with the name `propertyName` | true     | -             | {"catalog.uri": "http://localhost:8181"}                                                                                                                                                                   |
```
Comment (maha-hajja) [inline README.md:27]: could we also move this param to the end of the table, since it's the only "open" one.

Comment (hariso) [reply README.md]: Done!

File: README.md [hunk 3]
```diff
@@ -3,12 +3,71 @@
...
+| `s3.region`            | S3 region                                           | true     | -             | "us-east-1"                                                                                                                                                                                                        |
+| `catalog.catalog-impl` | Catalog implementation to be used                   | true     | -             | `"org.apache.iceberg.rest.RESTCatalog"` <br/> Possible values: <br/> - `"org.apache.iceberg.hadoop.HadoopCatalog"` <br/> - `"org.apache.iceberg.jdbc.JdbcCatalog"` <br/> - `"org.apache.iceberg.rest.RESTCatalog"` |
+| `catalog.propertyName` | Set a catalog property with the name `propertyName` | true     | -             | {"catalog.uri": "http://localhost:8181"}                                                                                                                                                                           |
```
Comment (maha-hajja) [inline README.md:36]:
```suggestion
| `catalog.{propertyName}` | Set a catalog property with the name `propertyName` | true     | -             | {"catalog.uri": "http://localhost:8181"}                                                                                                                                                                           |
```

**Review by maha-hajja — APPROVED** [review]

**Pull request merged**